### PR TITLE
Fix multiplication of double and int

### PR DESCRIPTION
**Pull request by ErikPitko**
I noticed that keyboard backlight doesn't change correctly. This change fixed it for me.

### CODE REVIEW

**Comment by FedeDP** [comment]
Hi! Thanks for opening this pr!
I am currently on vacation; I'll have a look at it next week, when I'm back :)

**Comment by FedeDP** [comment]
Hi!
I saw the PR; i think i found the issue and it is quite a nice issue you got hit by!
Thanks for finding it!

Can you test with this patch instead:
```
-        if (call(NULL, NULL, &kbd_args, "i", state.current_kbd_pct * max_kbd_backlight) == 0) {
+        const int new_kbd_pct = state.current_kbd_pct * max_kbd_backlight;
+        if (call(NULL, NULL, &kbd_args, "i", new_kbd_pct) == 0) {
```

Thanks!
Btw i'll review your PR anyway.

File: src/modules/backlight.c [hunk 1]
```diff
@@ -151,7 +151,9 @@ static void set_keyboard_level(const double level) {
          * on low ambient brightness, it must be turned on
          */
         state.current_kbd_pct = 1.0 - level;
-        if (call(NULL, NULL, &kbd_args, "i", state.current_kbd_pct * max_kbd_backlight) == 0) {
+	double new_value = state.current_kbd_pct * max_kbd_backlight;
```
Comment (FedeDP) [inline src/modules/backlight.c]: Indentation

File: src/modules/backlight.c [hunk 2]
```diff
@@ -151,7 +151,9 @@ static void set_keyboard_level(const double level) {
...
+	new_value += 0.5 - (new_value<0); // rounding
```
Comment (FedeDP) [inline src/modules/backlight.c]: Indentation and spacing

Comment (FedeDP) [reply src/modules/backlight.c]: Btw what is this needed for? I am just curious!

**Comment by FedeDP** [comment]
The issue is, quite stupidly, that "call()" is a variadic function; in my previous code, i was actually passing a double to it (ie: 8B long), and then only attaching first 4B to dbus call.
As it is a variadic function, the double i was passing as parameter was not automatically downcasted to int, obviously.
Quite a stupid oversight, but very interesting.

File: src/modules/backlight.c (hunk 2 above)
Comment (ErikPitko) [reply src/modules/backlight.c]: if you mean the new_value variable, it's just for rounding to the closest int, so it won't just cut all decimals when converting to int. As the command for changing brightness takes only int values. It's just nice to have imo.

Comment (FedeDP) [reply src/modules/backlight.c]: Ok, I see. I meant, i understood the simple algorithm; it was more about why not using [round](https://linux.die.net/man/3/round) instead. 
I did not understand if there was any reason behind that!

File: src/modules/backlight.c [hunk 3]
```diff
@@ -151,8 +151,8 @@ static void set_keyboard_level(const double level) {
          * on low ambient brightness, it must be turned on
          */
         state.current_kbd_pct = 1.0 - level;
-	double new_value = state.current_kbd_pct * max_kbd_backlight;
-	new_value += 0.5 - (new_value<0); // rounding
+        double new_value = state.current_kbd_pct * max_kbd_backlight;
+        new_value += 0.5 - (new_value<0); // for rounding to the closest int value
```
Comment (FedeDP) [inline src/modules/backlight.c]: Can we use https://linux.die.net/man/3/round instead?

Comment (FedeDP) [reply src/modules/backlight.c]: Please note that new_value will never be negative.

**Comment by FedeDP** [comment]
After this small improvement, i think this is ready to get merged, thank you!

File: src/modules/backlight.c (hunk 2 above)
Comment (ErikPitko) [reply src/modules/backlight.c]: Oh, I tried to make it so we won't require math.h. Didn't notice that it was already included. I will fix it.

**Comment by FedeDP** [comment]
Thank you very much!

**Pull request merged**